src/sentry/api/endpoints/project_group_index.py

```diff
@@ -230,11 +230,10 @@ def _search(self, request, project, extra_query_kwargs=None):
             query_kwargs.update(extra_query_kwargs)
 
         try:
-            if features.has('organizations:environments', project.organization, actor=request.user):
-                query_kwargs['environment'] = self._get_environment_from_request(
-                    request,
-                    project.organization_id,
-                )
+            query_kwargs['environment'] = self._get_environment_from_request(
```

Comment:
I left this in with the frontend changes as without it customers not in the environment feature would see the ux but get inconsistent server behavior.

Reply:
Gotcha. Forgot we haven't turned this on for everyone yet, would be great to do that before merging this PR so these code paths never get executed. (https://github.com/getsentry/getsentry/blob/master/getsentry/features.py)

Reply:
Is that a better path than removing all the checks and then removing the feature from getsentry?

Reply:
I think so. I'm not sure if this is excessively cautious but it would prevent weird situations like this. You could flip the default to true in Sentry by changing this line to true https://github.com/getsentry/sentry/blob/master/src/sentry/conf/server.py#L774

Reply:
Agree with lyn, I think we usually flip the feature tag on for everyone in getsentry, and then follow up with removing any feature flagged code in sentry.

Reply:
Ok. I will go down that path as well.

Reply:
Feature flipped on in #9769. The changes in #9742 and #9753 remove feature gated code from sentry and sentry/getsentry#2200 removes the feature gate in getsentry.

I apologize for approaching this backwards of how I should have :blush: